### PR TITLE
Display function signature on ctrl+hover (fixes #1713)

### LINKED ISSUE #1713: Modify the CTRL+hover output to show function signature
This has bugged me for quite some time.  Here is an example of holding CTRL while hovering over a method/function in different languages:

Java:
image

Python:
image

Go:
image

As you can see with Java and Python, doing the CTRL hoverover will show what the parameters are to the method and what it might return.  With the Go plugin, all it shows is what file the function came from, which isn't nearly as useful.  This is is especially annoying in languages like go where you can do

    x := foo(y, z)

because I have no idea of telling what x is (is it an int? string?  interface{}?).

### PR DESCRIPTION
**Pull request by dlsniper**

### CODE REVIEW

**Comment by ignatov** [comment]
Hmm, will be cool to a have a test for this.

File: src/com/goide/GoDocumentationProvider.java [hunk 1]
```diff
@@ -149,11 +152,23 @@ else if (type != null) {
...
   private static String getParametersAsString(@NotNull GoParameters parameters) {
     return StringUtil.join(GoParameterInfoHandler.getParameterPresentations(parameters), ", ");
   }
+
+  @Nullable
+  @Override
+  public String getQuickNavigateInfo(PsiElement element, PsiElement originalElement) {
+    if (element instanceof GoNamedElement) {
+      String result = getSignature(element);
+      if (result != "") {
```
Comment (zolotov) [inline src/com/goide/GoDocumentationProvider.java]: StringUtil#nullize

File: src/com/goide/GoDocumentationProvider.java [hunk 2]
```diff
@@ -46,6 +46,9 @@ public String generateDoc(PsiElement element, PsiElement originalElement) {
       Collection<PsiElement> children = PsiTreeUtil.findChildrenOfType(topLevel, element.getClass());
       boolean alone = children.size() == 1 && children.iterator().next().equals(element);
       String result = getSignature(element);
+      if (result != "") {
+        result = "<p><b>"+result+"</b></p>\n";
```
Comment (zolotov) [inline src/com/goide/GoDocumentationProvider.java]: code style

File: src/com/goide/GoDocumentationProvider.java [hunk 3]
```diff
@@ -46,6 +46,9 @@ public String generateDoc(PsiElement element, PsiElement originalElement) {
       Collection<PsiElement> children = PsiTreeUtil.findChildrenOfType(topLevel, element.getClass());
       boolean alone = children.size() == 1 && children.iterator().next().equals(element);
       String result = getSignature(element);
+      if (result != "") {
```
Comment (zolotov) [inline src/com/goide/GoDocumentationProvider.java]: do not use != for comparing objects

Comment (dlsniper) [inline src/com/goide/GoDocumentationProvider.java]: Force of habit from comparing strings in Go (and lack of practice with Java)

File: src/com/goide/GoDocumentationProvider.java (hunk 1 above)
Comment (dlsniper) [inline src/com/goide/GoDocumentationProvider.java]: Not sure how to use StringUtil#nullize in this context

**Comment by dlsniper** [comment]
@ignatov I'd be happy to add those. Any example for this that I could have a look at? Also, is it too complex to make the parameter types link to their Go types like in the other languages?

**Comment by ignatov** [comment]
Merged manually

**Pull request closed without merging**